pkg/asset/manifests/azure/cluster.go

```diff
@@ -104,6 +104,39 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 	}
 
 	virtualNetworkID := ""
+	lbip := capz.DefaultInternalLBIPAddress
+	machineCidr := installConfig.Config.MachineNetwork
+	// Check if default lbip is within machine network.
+	isInMachineCIDR := false
+	for _, cidrRange := range machineCidr {
+		_, ipnet, err := net.ParseCIDR(cidrRange.CIDR.String())
+		if err != nil {
+			return nil, fmt.Errorf("failed to get machine network CIDR: %w", err)
+		}
+		if ipnet.Contains(net.ParseIP(lbip)) {
+			isInMachineCIDR = true
+			break
+		}
```

Comment:
`cidrRange.CIDR` is already an `ipnet.IPNet` so you can simplify by skipping the conversion

```suggestion
		if cidrRange.CIDR.Contains(net.ParseIP(lbip)) {
			isInMachineCIDR = true
			break
		}
```